src/main/scala/za/co/absa/commons/spark/SchemaUtils.scala

```diff
@@ -200,4 +210,25 @@ object SchemaUtils {
 
     diff.map(_.stripPrefix("."))
   }
+
+  /**
+   * Checks if the originalSchema is a subset of subsetSchema.
+   *
+   * @param subsetSchema The schema that needs to be extracted
+   * @param originalSchema The schema that needs to have at least all t
+   * @return true if provided schemas are the same ignoring nullability
+   */
+  def isSubset(subsetSchema: StructType, originalSchema: StructType): Boolean = {
+    val fields1 = getMapOfFields(subsetSchema)
+    val fields2 = getMapOfFields(originalSchema)
+
+    fields1.values.foldLeft(true)((stillSame, field1) => {
```

Comment:
`fields1.keys.forall(...)` using `keys` you get the lowercase name for free and `forall` seems better fit here.

I would also check subsetSchema.size <= originalSchema.size

Reply:
I did most of this, but I skipped the `subsetSchema.size <= originalSchema.size` since there is much more depth to schemas testing, not sure how much it would improve the speed vs how harder to read it would be. 

I have an example here. You tell me if it is something you would like to see in there or not.

```scala
(subsetFields.size <= originalFields.size) && subsetFields.forall( subsetField =>
  originalFields.contains(subsetField._1) &&
    equivalentTypes(subsetField._2.dataType, originalFields(subsetField._1).dataType) )
```

Reply:
ok, you are right, if the data are not too big it's probably not worth it.